src/util/runtimeloggingcategory.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <QByteArray>
+#include <QLoggingCategory>
+
+/// RuntimeLoggingCategory is a wrapper around QLoggingCategory
+/// that facilitates using a QLoggingCategory for a category
+/// name string that is determined at run time. To use a
+/// RuntimeLoggingCategory, simply pass it to the
+/// qCInfo, qCDebug, qCWarning, or qCCritical macros like a
+/// QLoggingCategory. If the category name is fully known at
+/// compile time, simply construct a QLoggingCategory with a
+/// string literal instead of using RuntimeLoggingCategory.
+///
+/// RuntimeLoggingCategory stores the logging category name as a
+/// QByteArray. This is helpful because QLoggingCategory has a deleted
+/// copy constructor, does not take ownership nor make a copy of the
+/// const char* passed to its constructor, and requires that the
+/// data pointed to by the const char* lives as long as the
+/// QLoggingCategory.
+///
+/// RuntimeLoggingCategory takes advantage of QByteArray's
+/// implicit sharing to ensure that the const char* for the
+/// QLoggingCategory lives as long as the QLoggingCategory without
+/// requiring excessive deep copies.
+struct RuntimeLoggingCategory {
+  public:
+    RuntimeLoggingCategory(const QByteArray& oldByteArray)
```

Comment:
- `explicit`
- `oldByteArray` -> `category`
- Pass QByteArray by value and use `std::move` for initializing the member